spec/gollum_git_repo_modification_spec.rb

```diff
@@ -0,0 +1,157 @@
+require 'spec_helper'
+require 'rspec/expectations'
+
+RSpec::Matchers.define :a_blob_named do |expected|
+  match do |actual|
+    actual.is_a?(Gollum::Git::Blob) && actual.name == expected
+  end
+end
+
+describe Gollum::Git::Repo do
+
+  after(:each) do
+    if parent_commit != repo.head.commit
+      repo.update_ref(repo.head.name, parent_commit.id)
+    end
+  end
+
+  let(:repo) { Gollum::Git::Repo.new(fixture('dot_bare_git'), :is_bare => true) }
+
+  let(:index) do
+    repo.index.read_tree(repo.head.commit.tree.id)
+    repo.index
+  end
+
+  let(:actor) { Gollum::Git::Actor.new("Tom Werner", "tom@example.com") }
+
+  let(:message) { 'Test commit' }
+
+  let(:parent_commit) { repo.head.commit }
+
+  def commit_index
+    index.commit(message, [parent_commit], actor, nil, repo.head.name)
+  end
+
+  describe "#head#commit" do
+
+    context "after committing" do
+
+      before(:each) do
+        index.add('Add.txt', 'Some data')
+        commit_index
```

Comment:
I think having two spec suits (Rspec and Test::Unit) in gollum-lib would 
complicate things a bit too much. Best case scenario would be to have 
all the gollum-lib tests in Rspec and then include the adapter tests, 
but that's probably not going to ever happen. :-) So I think we should 
stick with the poor man's interface for now, and just copy some of the 
repo-setup code from the gollum-lib test helpers. See 
https://github.com/gollum/gollum-lib/blob/master/test/helper.rb#L45-L66